templates/types/llamaindexserver/fastapi/gen_ui.py

```diff
@@ -0,0 +1,594 @@
+    for event_type, events in events_by_type.items():
+        # Add schema as an event
+        result_events.append(json.loads(events[0].schema_json()))
+
+        # Add some sample for the event
+        num_samples = min(5, len(events))
+        samples = random.sample(events, num_samples)
+        for sample in samples:
+            result_events.append(sample.model_dump())
+
+    return result_events
+
+
+async def execute_workflow_and_filter_events(
+    workflow_input: str, expected_events: List[str], force_refresh: bool = False
+) -> List[Dict[str, Any]]:
+    """
+    Execute the workflow and filter events based on expected event types.
+    Returns a list containing both event schemas and sample events.
+
+    Args:
+        workflow_input: The input for the workflow
+        expected_events: List of event types to filter for
+        force_refresh: If True, ignores cached results and forces a new workflow execution
+    """
+    console = Console()
+
+    # Check cache file if not forcing refresh
+    if not force_refresh and os.path.exists(CACHE_FILE):
+        try:
+            with open(CACHE_FILE, "r") as f:
+                cached_events = json.load(f)
+                console.print(
+                    Panel(
+                        "[bold cyan]Using cached results[/bold cyan] for this workflow input. \nTo force a fresh execution, use the [bold yellow]--force-refresh[/bold yellow] flag.",
+                        title="🚀 Cache Hit",
+                        border_style="cyan",
+                    )
+                )
+                return cached_events
+        except json.JSONDecodeError:
+            # If cache file is corrupted, ignore it
+            pass
+
+    # Execute workflow and filter events
+    workflow_events = await execute_workflow(workflow_input)
+    result_events = filter_events(workflow_events, expected_events)
+
+    # Cache the results
+    try:
+        with open(CACHE_FILE, "w") as f:
+            json.dump(result_events, f, indent=2)
+    except Exception as e:
+        console.print(f"[yellow]Warning:[/yellow] Failed to cache results: {e}")
+
+    return result_events
+
+
+async def main(
+    workflow_input: str,
+    events: List[str],
+    output_file: str,
+    force_refresh: bool = False,
+):
+    from llama_index.llms.anthropic import Anthropic
+
+    console = Console()
+
+    # Execute workflow and get filtered events with schema
+    console.rule("[bold blue]Step 1: Execute Workflow[/bold blue]")
+    filtered_events = await execute_workflow_and_filter_events(
+        workflow_input=workflow_input,
+        expected_events=events,
+        force_refresh=force_refresh,
+    )
+
+    # Generate UI components
+    console.rule("[bold blue]Step 2: Generate UI Components[/bold blue]")
+    llm = Anthropic(model="claude-3-7-sonnet-latest", max_tokens=8192)
+    workflow = GenUIWorkflow(llm=llm, timeout=500.0)
+    await workflow.run(events=filtered_events, output_file=output_file)
```

Comment:
```suggestion
    await workflow.run(events=frontend_events, output_file=output_file)
```

get_frontend_events should not return the source code string, but a list of the event classes. the advantage is that the user of create-llama can then also do:
```suggestion
   from app.workflow import FrontendEvent
   await workflow.run(events=[FrontendEvent], output_file=output_file)
```

Reply:
if possible in the first version i would just skip implementing `get_frontend_events` and assume that the event is called `FrontendEvent` in the workflow